packages/liveblocks-core/src/api-client.ts

```diff
@@ -1526,6 +1546,14 @@ export function createApiClient<M extends BaseMetadata>({
       }
     );
 
+    const groups = json.groups.map(convertToGroupData);
+
+    // Instead of being returned publicly, the user's groups are put in
+    // a separate store which is also used for on-demand fetching.
+    for (const group of groups) {
+      groupsStore.fill(group.id, group);
```

Comment:
Since `.fill()` calls `update` which in turn set a signal, this loop means the same signal might get updated multiple times (triggering a notification waterfall after every update potentially, depending on configrued listeners). A best practice is to wrap such updates in a `batch()` which will only trigger a single notification waterfall at the end.

You could wrap it right here, or you could make `.fill()` take an array instead and do the batch wrapping there (closer to the actual signal), which I think would be nicest to avoid bleeding that complexity in here.

Reply:
Great catch, since this API is internal (and new for this use-case) I'll improve it to support arrays 👍